### PR TITLE
WIP Separate JLSO into its own project

### PR DESCRIPTION
**Pull request by fchorney**

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Documentation: https://github.com/JuliaCI/Appveyor.jl
+environment:
+  matrix:
+  - julia_version: 0.6
```
Comment (iamed2) [inline .appveyor.yml]: List items need another indent here

**Comment by fchorney** [comment]
0.6 error seems to do with https://github.com/invenia/JLSO.jl/blob/master/src/JLSO.jl#L111 looks like it defines `:compat` instead of `:compact`, and if you do change it to `:compact` then it just becomes an infinite loop. I figured I would raise an issue about that after this was merged in.

The nightly test seems to be related to https://github.com/samoconnor/LazyJSON.jl/issues/5

File: .appveyor.yml (hunk 1 above)
Comment (fchorney) [reply .appveyor.yml]: ah yes, good catch

File: README.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# JLSO
+
+[![Stable](https://img.shields.io/badge/docs-stable-blue.svg)](https://invenia.github.io/JLSO.jl/stable)
+[![Dev](https://img.shields.io/badge/docs-dev-blue.svg)](https://invenia.github.io/JLSO.jl/dev)
+[![Build Status](https://travis-ci.com/invenia/JLSO.jl.svg?branch=master)](https://travis-ci.com/invenia/JLSO.jl)
+[![Build Status](https://ci.appveyor.com/api/projects/status/github/invenia/JLSO.jl?svg=true)](https://ci.appveyor.com/project/invenia/JLSO-jl)
+[![Codecov](https://codecov.io/gh/invenia/JLSO.jl/branch/master/graph/badge.svg)](https://codecov.io/gh/invenia/JLSO.jl)
```
Comment (oxinabox) [inline README.md]:
```suggestion
[![Codecov](https://codecov.io/gh/invenia/JLSO.jl/branch/master/graph/badge.svg)](https://codecov.io/gh/invenia/JLSO.jl)

JLSO wraps other formats (currently [BSON](https://github.com/MikeInnes/BSON.jl)),
while storing key metadata suchs as the version of julia and all involved packages
that were used in creating the file.
```

**Pull request merged**